pkg/errorprinter/errorprinter.go

```diff
@@ -0,0 +1,83 @@
+package errorprinter
+
+import (
+	"strings"
+	"unicode"
+
+	"github.com/fatih/color"
+)
+
+const (
+	prefix = "Error: "
+)
+
+type ErrorPrinter struct {
+	DisableColors bool
+}
+
+func New() *ErrorPrinter {
+	ep := &ErrorPrinter{}
+
+	return ep
+}
+
+func (ep *ErrorPrinter) Format(pErr error) string {
+	if pErr == nil {
+		return ""
+	}
+
+	errorRows := strings.Split(pErr.Error(), "\n")
+
+	var messageBuilder strings.Builder
+
+	{
+		title := ep.formatTitle(errorRows[0])
+		messageBuilder.WriteString(title)
+		messageBuilder.WriteString("\n")
+	}
+
+	{
+		// Add subtitle, if it exists.
+		for _, row := range errorRows[1:] {
+			messageBuilder.WriteString(row)
+			messageBuilder.WriteString("\n")
+		}
+	}
+
+	return messageBuilder.String()
+}
+
+func (ep *ErrorPrinter) formatTitle(title string) string {
```

Comment:
They way we write error messages will make it very hard to debug if the thing before `:` and stack is not printed. I fully agree for some kinds we shouldn't print stack. Like for `invalidFlagError` which I'm trying to do [here](https://github.com/giantswarm/devctl/pull/67/files#diff-7c5bc5c38f66a3679f1d4ad81c524841R41-R43).

I'm not blocking this because I understand the output is really ugly for the end user but I also think debugging may become hard.

I think it might be an idea to invest some time into something like `microerror.Pretty` which would also print stacks in a digestable way + handling `invalidFlagError` separately.

We can also agree on some `userError` kind which would be printed the same way `invalidFlagError` is printed.

Reply:
Adding some extra metadata to each microerror is definitely something I'll look into. Also, some built-in error types (like the `invalidFlagError`).

For now, I'll go with having a `--debug` flag (a separate PR will come) for viewing full error output